roqoqo-iqm/src/interface.rs

```diff
@@ -124,6 +139,23 @@ pub fn call_circuit<'a>(
                 number_measurements = *o.number_measurements();
                 let readout = o.readout().clone();
 
+                match o.qubit_mapping() {
+                    None => {
+                        register_mapping.insert(
+                            o.readout().to_string(),
+                            (0..device_number_qubits).collect(),
+                        );
+                    }
+                    Some(map) => {
+                        for qubit in map.keys().sorted() {
+                            register_mapping
+                                .get_mut(o.readout())
+                                .unwrap()
```

Comment:
Could this fail? Maybe a match statement would be safer

Reply:
It does allow this, Circuit doesn't do any checking. Also, there could be a typo (for instance from copy-pasting) in one of the registers, and this could lead to errors

Reply:
Got it